docs/framework/guides/tutorials/using-react-in-a-widget.md

```diff
@@ -0,0 +1,1285 @@
+					'tableRow',
+					'mergeTableCells'
+				]
+			},
+			// The configuration of the Products plugin. It specifies a function that will allow
+			// the editor to render a React <ProductPreview> component inside a product widget.
+			products: {
+				productRenderer: ( id, domElement ) => {
+					const product = this.props.products.find( product => product.id === id );
+
+					ReactDOM.render(
+						<ProductPreview id={id} {...product} />,
+						domElement
+					);
+				}
+			}
+		};
+
+		this.handleEditorDataChange = this.handleEditorDataChange.bind( this );
+		this.handleEditorInit = this.handleEditorInit.bind( this );
+	}
+
+	// A handler executed when the user types or modifies the editor content.
+	// It updates the state of the application.
+	handleEditorDataChange( evt, editor ) {
+		this.setState( {
+			editorData: editor.getData()
+		} );
+	}
+
+	// A handler executed when the editor has been initialized and is ready.
+	// It synchronizes the initial data state and saves the reference to the editor instance.
+	handleEditorInit( editor ) {
+		this.editor = editor;
+
+		this.setState( {
+			editorData: editor.getData()
+		} );
+
+		// CKEditor 5 inspector allows you to take a peek into editor's model and view
+		// data layers. Use it to debug the application and learn more about the editor.
+		CKEditorInspector.attach( editor );
+	}
+
+	render() {
+		return [
+			// The application renders two columns:
+			// * in the left one, the <CKEditor> is rendered and the textarea displaying
+			//   live editor data.
+			// * in the right column, a <ProductList> is rendered with available <ProductPreviews>
+			//   to choose from.
+			<div className="app__offer-editor" key="offer-editor">
+				<h3>Product offer editor</h3>
+				<CKEditor
+					editor={ClassicEditor}
+					data={this.state.editorData}
+					config={this.editorConfig}
+					onChange={this.handleEditorDataChange}
+					onInit={this.handleEditorInit}
+				/>
+
+				<h3>Editor data</h3>
+				<textarea value={this.state.editorData} readOnly={true}></textarea>
+			</div>,
+			<ProductList
+				key="product-list"
+				products={this.props.products}
+				onClick={( id ) => {
+					this.editor.execute( 'insertProduct', id );
+					this.editor.editing.view.focus();
+				}}
+			/>
+		];
+	}
+}
+```
+
+The JavaScript code is ready but to run the application we need to specify a couple of product definitions. Let us do that when mounting the `<App>` component:
+
+```js
+// app.js
+
+class App extends React.Component {
+	// ...
+}
+
+// Render the <App> in the <div class="app"></div> element found in DOM.
+ReactDOM.render(
+	<App
+		// Feeding the application with pre–defined products.
+		// In a real-life application, this sort of data would be loaded
+		// from a database. To keep this tutorial simple, we're using
+		// a couple of hard–coded product definitions.
+		products={[
+			{
+				id: 1,
+				name: 'Colors of summer in Poland',
+				price: '$1500',
+				image: 'product1.jpg'
```

Comment:
Must the path be relative to the CSS file?

Reply:
Not sure what's the problem here. Does not it work the way it is now?

Reply:
Everything works fine. 

I just wanted to know why `assets/product1.jpg` which is a relative path to `index.html` didn't work. Then I figured out that the may the path must be relative to CSS file.